modules/casper.js

```diff
@@ -2745,7 +2748,39 @@ function createPage(casper) {
     };
     page.onPageCreated = function onPageCreated(popupPage) {
         casper.emit('popup.created', popupPage);
-        popupPage.onLoadFinished = function onLoadFinished() {
+        if (casper.options.pageSettings.userAgent !== defaultUserAgent) {
+            popupPage.customHeaders = {"User-Agent": casper.options.pageSettings.userAgent};
+        }
+        popupPage.settings = utils.mergeObjects(popupPage.settings, casper.options.pageSettings);
+        if (utils.isClipRect(casper.options.clipRect)) {
+            popupPage.clipRect = casper.options.clipRect;
+        }
+        if (utils.isObject(casper.options.viewportSize)) {
+            popupPage.viewportSize = casper.options.viewportSize;
+        }
+        ['onAlert', 'onCallback', 'onConfirm', 'onConsoleMessage',
+        'onError', 'onFilePicker', 'onInitialized', 'onClosing',
+        'onNavigationRequested', 'onPrompt', 'onResourceReceived',
+        'onResourceError',
+        // 'onPageCreated',
```

Comment:
events are all defined . Few events are identical and only 4 events are redefined.